subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache
+
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.FileVisitOption
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Stream
+
+import static org.gradle.initialization.IGradlePropertiesLoader.ENV_PROJECT_PROPERTIES_PREFIX
+import static org.gradle.configurationcache.EnvironmentVarKeySource.GRADLE_ENCRYPTION_KEY_ENV_KEY
+import static org.gradle.util.Matchers.containsLine
+import static org.gradle.util.Matchers.matchesRegexp
+
+class ConfigurationCacheEncryptionIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
+    String encryptionKeyText
+    String encryptionKeyAsBase64
+
+    def setup() {
+        encryptionKeyText = "01234567890123456789012345678901"
+        encryptionKeyAsBase64 = Base64.encoder.encodeToString(encryptionKeyText.getBytes(StandardCharsets.UTF_8))
+    }
+
+    def "configuration cache can be loaded without errors if encryption is #status, #encryptionTransformation"() {
+        given:
+        def additionalOpts = [
+            "-Dorg.gradle.configuration-cache.internal.encryption-alg=${encryptionTransformation}"
```

Comment:
❓ Is this meant to be the user interface?

Reply:
I can drop it but was thinking of keeping it (undocumented) at least for testing

Reply:
Please see this test for the build options

https://github.com/gradle/gradle/pull/24150/files#diff-406b5abd6fdd148ae586d78a9f2d60819f4e7dce3481b9cc3e605538c65603d7R75